dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -133,27 +133,52 @@ internal KernelFunction(string name, string description, IReadOnlyList<KernelPar
             cancellationToken.ThrowIfCancellationRequested();
 
             // Invoke pre-invocation event handler. If it requests cancellation, throw.
-            if (kernel.OnFunctionInvoking(this, arguments)?.Cancel is true)
+            var invokingEventArgs = kernel.OnFunctionInvoking(this, arguments);
+
+            // Invoke pre-invocation filter. If it requests cancellation, throw.
+            var invokingContext = kernel.OnFunctionInvokingFilter(this, arguments);
```

Comment:
Actually, this is a correct place of `invokingContext`, because if I move it a little bit lower and event hook will trigger cancellation, the pre-invocation filter won't be triggered in this case, and I believe it's incorrect approach. In current implementation, we trigger both event hook and filter, and only after that we use event args and context objects to process result:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/e70e39df-50af-4588-b862-7f6d40e7afe6)